components/script/dom/resizeobserver.rs

```diff
@@ -138,7 +138,7 @@ impl ResizeObserver {
             // The spec plans to store multiple reported sizes,
             // but for now there can be only one.
             observation.last_reported_sizes.borrow_mut()[0] = size_impl;
-            *observation_state = ObservationState::Done;
+            *observation.state.borrow_mut() = ObservationState::Done;
```

Comment:
This won't work—the borrow() from earlier is still active, so this will conflict and cause a panic.

Reply:
I've tried enclosing the `borrow()` within a block so that it's no longer active when we borrow mutably but the tests still seem to be crashing. What else can be done here? 🤔

Reply:
Okay, I've refactored the code like you recommended, lmk if there's more room for improvements :)